src/v/cluster/node/local_monitor.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/node/local_monitor.h"
+
+#include "cluster/node/types.h"
+#include "config/node_config.h"
+#include "model/timestamp.h"
+#include "version.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/file.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/reactor.hh>
+
+#include <chrono>
+#include <seastarx.h>
+
+namespace cluster::node {
+
+// TODO make async
+ss::future<> local_monitor::update_state() {
+    auto disks = get_disks();
+    auto vers = application_version((std::string)redpanda_version());
```

Comment:
avoid c-style casts (e.g. `(std::string)xxx`). here you can explicitly construct an sstring or if named_type forwards its constructor parameters (I'm not sure it does) you might be able to pass the char* all the way down.

Reply:
Cool. That was my inclination but this was existing line I just moved. I'll address in followup patch.